### DISCUSSION TITLE
ortools and protobuf-lite ?

### DISCUSSION THREAD

**Discussion by cquenelle** [google/or-tools#2443] [Build and configuration issues] [answered]
I'm using OR-Tools 7.8 and protobuf 3.12.2.

I'm trying to make OR-Tools use protobuf-lite to get rid of the static initializers that it creates in the ortools library.  This is a performance issue for me.

If my project is not using the parts of OR-Tools that depend on non-lite pieces of protobuf, I'm wondering how hard it would be to disable them.

I'm looking at __PORTABLE_PLATFORM__ in the OR-Tools code base.  It seems to disable some external dependencies. But I haven't found any documentation for it yet. How is __PORTABLE_PLATFORM__ normally used?

Does anyone have any guidance to offer before I start down this road?

Chris

**Comment by lperron** [comment, marked as answer]
Basically, if you define __PORTABLE_PLATFORM__, it disable non proto-lite usage in port/proto_utils.h

I am not sure you want to define __PORTABLE_PLATFORM__ everywhere.

**Reply by lperron** [reply]
On master, I have pushed both portable and non portable versions of the code in the ortools/port/ directory.

If you compile with '__PORTABLE_PLATFORM__' defined, it should only use proto-lite compatible APIs.

**Reply by cquenelle** [reply]
Thanks for the info, I figured out my issue.
